executor/read/immediate_executor.rs

```diff
@@ -395,7 +388,7 @@ impl IntersectionExecutor {
 
         let input_row = self.input.as_mut().unwrap().peek().unwrap().as_ref().map_err(|&err| err.clone())?;
         for &position in &self.outputs_selected {
-            if position.as_usize() < input_row.len() {
+            if position.as_usize() < input_row.len() && !input_row.get(position).is_empty() {
```

Comment:
Wait, is there ever a case where this check can fail?

Reply:
Yes. In a disjunction, one branch can set row positions as A: 0, B: 1, C: 2, but the other branch may choose traversal such that it produces the vvariables in the order A, C, B. In that case, when we get to the last step, the input will be `0: A, 1: None, 2: C`, and the `1: B` assigned by the iterator will be (attempted to be) overwritten by the `1: None` from the input.